e2e/tests/nightly/admin/lock-workspace.spec.ts

```diff
@@ -0,0 +1,138 @@
+    await workspacesPage.load();
+    await workspacesPage.createWorkspace(workspaceName);
+    const dataPage = new WorkspaceDataPage(page);
+    await dataPage.waitForLoad();
+    //extract the Workspace-Namespace
+    workspaceNamespace = await dataPage.extractWorkspaceNamespace();
+    //create the dataset and notebook
+    await createDatasetNotebook(page, pyNotebookName);
+    await navigation.navMenu(page, NavLink.WORKSPACE_ADMIN);
+    const workspaceAdminPage = new WorkspaceAdminPage(page);
+    await workspaceAdminPage.waitForLoad();
+    await workspaceAdminPage.getWorkspaceNamespaceInput().type(workspaceNamespace);
+    await workspaceAdminPage.clickLoadWorkspaceButton();
+    await workspaceAdminPage.waitForLoad();
+    //click on the LOCK WORKSPACE button
+    const lockWorkspaceModal = await workspaceAdminPage.clickLockWorkspaceButton(workspaceStatus.Lock);
+    expect(await lockWorkspaceModal.getLockWorkspaceButton().isCursorNotAllowed()).toBe(true);
+    await lockWorkspaceModal.createLockWorkspaceReason(reasonText);
+    //const lockWorkspaceReason = await lockWorkspaceModal.createLockWorkspaceReason();
+    expect(await lockWorkspaceModal.getLockWorkspaceButton().isCursorNotAllowed()).toBe(false);
+    await lockWorkspaceModal.clickModalLockWorkspace();
+    await workspaceAdminPage.waitForLoad();
+    // verify the button now displays label "UNLOCK WORKSPACE"
+    expect(workspaceAdminPage.getLockWorkspaceButton(workspaceStatus.Unlock));
+    //await new WorkspacesPage(page).load();
+    await new HomePage(page).load();
+    const workspaceCard = await WorkspaceCard.findCard(page, workspaceName);
+    const lockedIcon = workspaceCard.getWorkspaceLockedIcon();
+    expect(lockedIcon).toBeTruthy();
+    await workspaceCard.verifyLockedWorkspaceMenuOptions();
+    await workspaceCard.clickLockedWorkspaceName(true);
+    const aboutPage = new WorkspaceAboutPage(page);
+    await aboutPage.waitForLoad();
+    // verify the workspace-locked banner displays
+    const aboutLockReason = await aboutPage.extractReasonMessage();
+    expect(aboutLockReason).toEqual(reasonText);
+    const aboutLockedIcon = aboutPage.getAboutLockedWorkspaceIcon();
+    expect(aboutLockedIcon).toBeTruthy();
+    // verify DATA & ANALYSIS tabs are disabled
+    await aboutPage.openDataPage({ waitPageChange: false });
+    await aboutPage.openAnalysisPage({ waitPageChange: false });
+    // verify share button is disabled
+    expect(await aboutPage.getShareButton().isCursorNotAllowed()).toBe(true);
+    await aboutPage.verifyLockedWorkspaceActionOptions();
+    // verify the user is able to navigate to edit page to update the description
+    await aboutPage.getAboutEditIcon();
+    const workspaceEdit = new WorkspaceEditPage(page);
+    await workspaceEdit.waitForLoad();
+    // verify user is able to edit the locked workspace
+    workspaceEditedName = await workspaceEdit.fillOutRequiredDuplicationFields();
```

Comment:
@aweng98 Since I'm reusing this function `fillOutRequiredDuplicationFields()`, I suggest we update the name to editWorkspaceName.

Reply:
No, `fillOutRequiredDuplicationFields()` is fine.